File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    
    var answer = n;
    var lst=[]
    
    for(i=0; i<n+2;i++){
        lst.push(1)
    }
    console.log(lst)
    for(i of reserve){
        lst[i]=2
    }
    console.log(lst)
    
    for(i of lost){
        lst[i]-=1
    }
    console.log(lst)

    for(i=1; i<n+1;i++){
        if(lst[i]===0){
            
            
            if(lst[i-1]===2){
                lst[i-1]-=1
                lst[i]=1
            } else if(lst[i+1]===2){
                lst[i+1]-=1
                lst[i]=1
                
            }
            
        }
    }
    
    
    for(i=1;i<n+1;i++){
        if(lst[i]==0){
            answer-=1
        }
    }
    
    
    
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer=n
    lst = [1 for i in range(n+2)]
    # print(lst)
    
    for i in reserve:
        lst[i]=2
    
    for i in lost:
        lst[i]-=1
    # print(lst)
    
    
    for i in range(1,len(lst)):
        if lst[i]==0:
            if lst[i-1]==2:
                lst[i-1]-=1
                lst[i]=1
            elif lst[i+1]==2:
                lst[i+1]-=1
                lst[i]=1
    
    for i in range(1,len(lst)):
        if lst[i]==0:
            answer -=1
        
                
            
    
    
        
    
    
    
    
    return answer